### PR TITLE
Feature/new conditional formatting7

### PR DESCRIPTION
**Pull request by OssianEPPlus**
- Added the ability to reference cells rather than values to all CFs that can do so in Excel
- Added ability to reference other worksheets on ConditionaFormattings in general where possible.
- Added Formula variable to ConditionalFormatting types wherever possible to set addresses with.
- Added several icons that were missing to iconsets (3Stars, 3Triangles, 5Boxes)
- Added ability to create CustomIconsets via Enum
- Added support for advanced Databar colors (Theme, Auto, Index, Tint properties) and databar support for negative value settings within epplus. Added ability to turn off or on border color, gradient and the set AxisPosition and color between negative and positive values
- Added extended support for Colourscale gradients and advanced colours (same as databar)
- Ensured better handling of Quotation Marks and special signs in Conditional Formattings handling text.
- Enabled multiple conditionalformattings on one address/range

- Restructured internal structure to be more performant

### CODE REVIEW

File: src/EPPlus/ConditionalFormatting/ExcelConditionalFormattingConstants.cs [hunk 1]
```diff
@@ -263,6 +264,13 @@ internal class Colors
       internal static readonly Color CfvoMiddleValue = Color.FromArgb(0xFF,0xFF,0xEB,0x84);
       internal static readonly Color CfvoHighValue = Color.FromArgb(0xFF,0x63,0xBE,0x7B);
     }
-    #endregion Colors
-  }
+
+        internal class DxfColor
```
Comment (JanKallman) [inline src/EPPlus/ConditionalFormatting/ExcelConditionalFormattingConstants.cs:268]: Seems empty, remove?

File: src/EPPlus/ConditionalFormatting/ColourScaleReadHandler.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+﻿using OfficeOpenXml.Utils.Extensions;
```
Comment (JanKallman) [inline src/EPPlus/ConditionalFormatting/ColourScaleReadHandler.cs:1]: Add license headers for all files.

File: src/EPPlus/ConditionalFormatting/ExcelConditionalFormattingCollection.cs [hunk 3]
```diff
@@ -1,1005 +1,1302 @@
...
-  and may only be used for noncommercial purposes 
-  https://polyformproject.org/licenses/noncommercial/1.0.0/
-
-  A commercial license to use this software can be purchased at https://epplussoftware.com
- *************************************************************************************************
-  Date               Author                       Change
- *************************************************************************************************
-  01/27/2020         EPPlus Software AB       Initial release EPPlus 5
- *************************************************************************************************/
+﻿using OfficeOpenXml.ConditionalFormatting.Contracts;
```
Comment (JanKallman) [inline src/EPPlus/ConditionalFormatting/ExcelConditionalFormattingCollection.cs:1]: Looks like the header has been removed here and in some other files.

File: src/EPPlus/ExcelRangeBase.cs [hunk 4]
```diff
@@ -1919,7 +1919,10 @@ public ExcelRangeBase Offset(int RowOffset, int ColumnOffset, int NumberOfRows,
         }
         /// <summary>
         /// Adds a new comment for the range.
-        /// If this range contains more than one cell, the top left comment is returned by the method.
+        /// If this range
```
Comment (JanKallman) [inline src/EPPlus/ExcelRangeBase.cs:1922]: Some new lines seems to be added here. Intended?

File: src/EPPlus/Utils/ExtentionFunctions.cs [hunk 5]
```diff
@@ -45,6 +45,19 @@ internal static string ToEnumString(this Enum enumValue)
             }
         }
 
+        //internal static T ToEnumThrowIfNull<T>(this string s) where T : struct
```
Comment (JanKallman) [inline src/EPPlus/Utils/ExtentionFunctions.cs:48]: Remove?

File: src/EPPlusTest/ConditionalFormatting/ConditionalFormattingTests.cs [hunk 6]
```diff
@@ -42,14 +54,146 @@ namespace EPPlusTest.ConditionalFormatting
     public class ConditionalFormattingTests : TestBase
```
Comment (JanKallman) [inline src/EPPlusTest/ConditionalFormatting/ConditionalFormattingTests.cs:54]: A lot of tests here. Might be a good idea to split this class into multiple classes.

**Review by JanKallman — CHANGES_REQUESTED** [review]

File: src/EPPlus/ConditionalFormatting/ExcelConditionalFormattingConstants.cs (hunk 1 above)
Comment (OssianEPPlus) [reply src/EPPlus/ConditionalFormatting/ExcelConditionalFormattingConstants.cs:268]: Yep. Also found a bunch of other references that only lead to classes that are not used anymore so I'm removing those too

File: src/EPPlus/ExcelRangeBase.cs (hunk 4 above)
Comment (OssianEPPlus) [reply src/EPPlus/ExcelRangeBase.cs:1922]: Nope. Unintentional by an aggressive sorting through search results at one point. Fixing.

File: src/EPPlus/Utils/ExtentionFunctions.cs (hunk 5 above)
Comment (OssianEPPlus) [reply src/EPPlus/Utils/ExtentionFunctions.cs:48]: yep

File: src/EPPlusTest/ConditionalFormatting/ConditionalFormattingTests.cs (hunk 6 above)
Comment (OssianEPPlus) [reply src/EPPlusTest/ConditionalFormatting/ConditionalFormattingTests.cs:54]: fair point. I considered it but at the time it was easier to just keep going and the point was partially to make a lot of the tests write to a single easy to overlook file. So those I will most likely make stay but a lot of tests are unthethered to that and mostly about colourscale, databar or Icons. I'll move that into separate classes

File: src/EPPlus/ConditionalFormatting/ExcelConditionalFormattingCollection.cs (hunk 3 above)
Comment (OssianEPPlus) [reply src/EPPlus/ConditionalFormatting/ExcelConditionalFormattingCollection.cs]: I've added it back in to all files and to new ones now

**Review by JanKallman — APPROVED** [review]

**Pull request merged**